frontend/src/components/worksheets/Worksheet/Worksheet.js

```diff
@@ -357,20 +342,47 @@ class Worksheet extends React.Component {
         this.togglePopupNoEvent(cmd_type);
     };
 
+    addCopyBundleRowsCallBack = (tableID, callback) => {
+        if (this.bundleTableID.has(tableID)) return;
+        this.bundleTableID.add(tableID);
+        this.copyCallBacks.push(callback);
+    };
+
     togglePopup = (cmd_type) => () => {
         if (cmd_type === 'deleteItem') {
             this.setState({ openDeleteItem: !this.state.openDeleteItem });
         }
         if (!this.state.showBundleOperationButtons) {
             return;
         }
-        const { openKill, openDelete, openDetach } = this.state;
+        const { openKill, openDelete, openDetach, openCopy } = this.state;
         if (cmd_type === 'rm') {
```

Comment:
Can you refactor this whole if/else statement to another function (as it gets copied verbatim later on)?